Automattic/woocommerce-payments
# New workflow to amend release changelog

Fixes #5126

#### Changes proposed in this Pull Request

- Add a new workflow `release-changelog.yml`
- Update the composite action `generate-changelog` to be able to either generate or amend the changelog entries, depending on a new input

The new workflow allows us to generate or amend the changelog entries and push the changes to a branch. I expect that this will only be used to amend the entries, but I still let the possibility to generate them as it didn't complicate things at all.
This can be used whenever one or several commits are added to the release branch after the official release code freeze, or during a patch release if needed.

#### Testing instructions

- Create and push a dummy branch to the repo, using that branch (so that it contains the new workflow, otherwise you won't be able to test it)
- Run the workflow to generate the changelog: `gh workflow run release-changelog.yml --ref <DUMMY_BRANCH> -f release-version="5.1.0" -f action-type="generate"`
- You should see a new commit on the dummy branch with the following changes:
  - All changelog entries under the `changelog` folder were inserted into `changelog.txt` under a new block for the release version provided, and those new entries were deleted from the folder too
  - The same changelog entries were pasted in `readme.txt`
  - The release version and release date are correct in both files w.r.t the inputs passed to the workflow (release date is set to "today" per default)

- Now, create one or several new changelog entries on the dummy branch with `npm run changelog`
- Run the workflow to amend the last changelog: `gh workflow run release-changelog.yml --ref >DUMMY_BRANCH> -f release-version="5.1.0" -f release-date="tomorrow" -f action-type="amend"`
- You should see a new commit on the dummy branch with the following changes:
  - All changelog entries under the `changelog` folder were inserted into the last changelog that was generated earlier, and those new entries were deleted from the folder too
  - The same new changelog entries were pasted in `readme.txt` under the last changelog that was generated earlier
  - The release version and release date are correct in both files w.r.t the inputs passed to the workflow (especially the date, which is intentionally different from the one we used to generate)

-------------------


**Post merge**
